File: backend/src/main/java/epidem/io/model/Simulation.java
package epidem.io.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import org.hibernate.annotations.Cascade;


import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
@Table(name = "simulation")
public class Simulation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Long populationSize;
    private Long infected;
    private Integer infectionRate;
    private Float mortalityRate;
    private Integer daysUntilRecovery;
    private Integer daysUntilDeath;
    private Integer simulationDays;

    @JsonIgnore
    @OneToMany(mappedBy = "simulation")
    List<Population> populations = new ArrayList<>();

}

File: frontend/src/app/simulation-element/simulation.ts
import { Identifiers } from "@angular/compiler";

export interface Simulation {
    id: number
    name: string
    populationSize: string
    infected: string
    infectionRate: string
    mortalityRate: string
    daysUntilRecovery: string
    daysUntilDeath: string
    simulationDays: string
}
